tests/test_children.py

```diff
@@ -13,92 +13,119 @@ def test_invalid_children_dict_not_array(self):
         """ Ensures InvalidTwine exceptions are raised when instantiating twines where `children` entry is incorrectly
         specified as a dict, not an array
         """
-        twine_file = self.path + "twines/invalid_children_dict_not_array_twine.json"
         with self.assertRaises(exceptions.InvalidTwine):
-            Twine(source=twine_file)
+            Twine(source={"children": {}})
 
     def test_invalid_children_no_key(self):
         """ Ensures InvalidTwine exceptions are raised when instantiating twines where a child
         is specified without the required `key` field
         """
-        twine_file = self.path + "twines/invalid_children_no_key_twine.json"
+        source = {
+            "children": [{"purpose": "The purpose.", "notes": "Here are some notes.", "filters": "tags:gis"}]
+        }
+
         with self.assertRaises(exceptions.InvalidTwine):
-            Twine(source=twine_file)
+            Twine(source=source)
 
     def test_valid_children(self):
-        """ Ensures that a twine can be instantiated with correctly specified children
+        """ Ensures that a twine with one child can be instantiated correctly.
         """
-        twine_file = self.path + "twines/valid_children_twine.json"
-        twine = Twine(source=twine_file)
-        self.assertEqual(len(twine._raw["children"]), 1)
+        source = {
+            "children": [{"key": "gis", "purpose": "The purpose.", "notes": "Some notes.", "filters": "tags:gis"}]
+        }
```

Comment:
In principle, I'm happy with the clear up of **most** (*not all*) of the files into fixtures within the tests themselves. As the test suite grows, clearly this'll make for faster execution.

However, I think the first thing for you to do is dig into `load_json` [here](https://github.com/octue/twined/blob/c3874db523b0a69dafeebc8a712c9ed54617b472/twined/utils/load_json.py#L41) and [here](https://github.com/octue/twined/blob/c3874db523b0a69dafeebc8a712c9ed54617b472/twined/utils/load_json.py#L52). Several points arise:

- ... the most important being that that `Twine(source="{}")` is *not the same* as `Twine(source={})` The latter is not roundtripped through JSON conversion (obviously to avoid the overhead of serialising/deserialising objects already in memory), so it's possible that object is not representable in JSON or deserialisable from it. This has the implication that test fixtures where JSON is incorrect will have to rely on the `"""{}""" notation to set up as strings.

- The different loaders in `load_json` (which are there for convenience for users to avoid writing the same kind of switching code themselves) still need to be covered. That can/should be done in a separate unit test case anyway.

- In the `load_json` function signature, you'll see that `kwargs` get passed down from the caller. This in principle allows the caller to set up custom decoders ([a decoder looks like this](https://github.com/octue/octue-sdk-python/blob/master/octue/utils/decoders.py)). At the moment there's no custom decoder used as default in twined but its worth being aware of this area in which  `Twine(source="{}")` could deviate even further from `Twine(source={})`. There's some tidying up to do in this encoding/decoding area because, for example, I want to remove [this horrible hack](https://github.com/octue/twined/blob/c3874db523b0a69dafeebc8a712c9ed54617b472/twined/twine.py#L161) that I made under time pressure, and resolve an [issue](https://github.com/octue/twined/issues/41) where I used a dumb name and dumb approach for decoding to actual objects.

Reply:
On the first bullet point, what do you think of this for the object case? 
```
elif not isinstance(source, str):
    logger.debug("Source is not a string, bypassing (returning raw data)")
    check("object")

    try:
        json.dumps(source)
        return source
    except TypeError:
        raise exceptions.invalid_json_map['twine']("Cannot load object - not JSON serialisable.")
```

That way, you could guarantee that `Twine(source="{}")` is the same as `Twine(source={})`

Reply:
For the second bullet point: I guess just write unit test needs to be written to ensure the JSON file case is covered, and then all other test cases could be moved into their tests?

Reply:
Ah I see, maybe the code I suggested above would be too rigid in that place then if other decoders are planned

Reply:
I have some questions about the `inbound` variable actually - I'll bring it up on Wednesday